src/Propel/Generator/Command/InitCommand.php

```diff
@@ -264,7 +264,7 @@ private function reverseEngineerSchema(OutputInterface $output, array $options)
         $outputDir = sys_get_temp_dir();
 
         $this->getApplication()->setAutoExit(false);
-        $fullDsn = sprintf('%s;user=%s;password=%s', $options['dsn'], urlencode($options['user']), urlencode($options['password']));
+        $fullDsn = sprintf('%s;user=%s;password=%s', $options['dsn'], $options['user'], $options['password']);
```

Comment:
@staabm PDO handles a password with a `;` by not including the password in the dsn.  (I think this answers your question... tell me if it doesn't.)

The function signature is:
`PDO::__construct ( string $dsn [, string $username [, string $password [, array $options ]]] )`
See
http://php.net/manual/en/pdo.construct.php

In fact, the dsn specification for for MySQL at
http://php.net/manual/en/ref.pdo-mysql.connection.php
does not include the password in the dsn string at all.

This is how Propel uses PDO internally as well.
https://github.com/propelorm/Propel2/blob/master/src/Propel/Runtime/Connection/PdoConnection.php#L47

Does that answer your question?